Prog/MAR_summary_units.sas

```diff
@@ -0,0 +1,146 @@
+/**************************************************************************
+ Program:  MAR_summary_units.sas
+ Library:  MAR
+ Project:  Urban-Greater DC
+ Author:   Rob and Yipeng
+ Created:  9/26/2018
+ Version:  SAS 9.4
+ Environment:  Local Windows session (desktop)
+ 
+ Description:  Create summary file of units from MAR address points. 
+
+ Modifications:
+**************************************************************************/
+
+%include "L:\SAS\Inc\StdLocal.sas";
+
+** Define libraries **;
+%DCData_lib( MAR )
+%DCData_lib( realprop )
+
+%let address_pt_date = 2018_06;
+%let revisions= New file;
+
+
+proc sort data = mar.address_points_&address_pt_date. out = addpt_in; by ssl; run;
+proc sort data = realprop.parcel_base out = parcel_in; by ssl; run;
+
+
+/* Set Address Points dataset from most recent update */
+data mar_units;
+	merge addpt_in (in=a)
+		  parcel_in (in=b where=(in_last_ownerpt=1));
+	by ssl;
+
+	/* If matched to parcel base, flag res and non-res units */
+	if b then do;
+		if ui_proptype in ("10","11","12","13","19") then res_flag = 1;
+			else res_flag = 0;
+	end;
+
+	/* Because condo units have individual SSLs, we have to assume
+	   a non-match to the address pt file is a condo unit */
+	if a and ui_proptype in (" ") then res_flag = 1;
+
+
+	/* Count number of units by property type*/
+	if res_flag = 1 then do;
+		total_res_units = ACTIVE_RES_OCCUPANCY_COUNT;
+		if ui_proptype in ("10") then total_sf_units = ACTIVE_RES_OCCUPANCY_COUNT;
+		if ui_proptype in ("13") then total_mfapt_units = ACTIVE_RES_OCCUPANCY_COUNT;
+		if ui_proptype in ("11"," ") then total_condo_units = ACTIVE_RES_OCCUPANCY_COUNT;
```

Comment:
@rpitingolo - technically condos could be in single family structures. I took out the multifamily bit. i'd be interested if you can think of how to make the distinction between the two types  - if you look at use code for example condo are distinguished between horizontal and vertical  see file:///L:\Libraries\Realprop\Doc\UseCode%20&%20UI_proptype.xls for the definitions of the usecodes.

Reply:
@lhendey couldn't we just use active_res_occupancy_count and if the value is >1 and condo then define it as a MF condo building?

Reply:
@rpitingolo no. single-family structures are not necessarily single-family.  Like the rowhouses in capital hill? they aren't getting coded as multifamily. I'm thinking about the rowhouses in columbia heights that are getting renovated an split into 3 unit condominiums.

Reply:
@lhendey I would consider a columbia heights row house with 3 condos to be multi-family though? Maybe we are splitting hairs and should just leave it as you edited.

Reply:
@rpitingolo yes it is a building with multiple families, but the structure type is not multifamily. it's fine not to make that distinction here.